testing/databaseTest.py

```diff
@@ -386,23 +387,23 @@ def general_check_metal_database_has_catalyst_properties(self, library):
             for element in 'CHON':
                 if not entry.binding_energies[element]:
                     raise KeyError('Entry {} has no {} binding energy'.format(entry.label, element))
-                if not isinstance(entry.binding_energies[element], tuple):
-                    raise TypeError('Entry {} binding energy value for {} should be a tuple, but is type {}'.format(
+                if not isinstance(entry.binding_energies[element], ScalarQuantity):
+                    raise TypeError('Entry {} binding energy value for {} should be a ScalarQuantity, but is type {}'.format(
                         entry.label, element, type(entry.binding_energies[element])))
-                if not isinstance(entry.binding_energies[element][0], float):
+                if not isinstance(entry.binding_energies[element].value, float):
                     raise TypeError('Entry {} binding energy for {} should be a float, but is type {}'.format(
-                        entry.label, element, type(entry.binding_energies[element][0])))
-                assert entry.binding_energies[element][0] < 0.  # binding energies should all be negative... probably
-                assert entry.binding_energies[element][1] == 'eV/molecule'
+                        entry.label, element, type(entry.binding_energies[element].value)))
+                assert entry.binding_energies[element].value < 0.  # binding energies should all be negative... probably
+                assert entry.binding_energies[element].units == 'eV/molecule'
 
             if not entry.surface_site_density:
                 raise AttributeError('Entry {} has no surface site density'.format(entry.label))
-            assert isinstance(entry.surface_site_density, tuple)
-            if not isinstance(entry.surface_site_density[0], float):
-                raise TypeError('Entry {} should be a float, but is type {}'.format(entry.label, type(entry.surface_site_density[0])))
-            assert 1e-8 > entry.surface_site_density[0] > 1e-10  # values should be reasonable
-            if not isinstance(entry.surface_site_density[1], str):
-                raise TypeError('Entry {} should be a str, but is type {}'.format(entry.label, type(entry.surface_site_density[1])))
+            assert isinstance(entry.surface_site_density, ScalarQuantity)
+            if not isinstance(entry.surface_site_density.value, float):
+                raise TypeError('Entry {} should be a float, but is type {}'.format(entry.label, type(entry.surface_site_density.value)))
+            assert 1e-8 > entry.surface_site_density.value > 1e-10  # values should be reasonable
```

Comment:
If you're ensuring the `.value` is in a certain range, then you've made an assumption about the units. Which you don't then check (besides ensuring it's a string).  Better to check that it is in fact a surface concentration, and then use the `value_si` if you want to check the value is about right.   At the moment,   `(1e-9, 'mol/cm^3')` would pass (it shouldn't), but `(1e-5, 'mol/m^2') would fail (it should pass).

Reply:
good idea, thanks